### PR TITLE
Rewrite the test cases for groups

### LINKED ISSUE #133: Group creation should be allowed with empty members list
## About

Group member list should not be edited through API. Adding 'read_only' to the serializer field should work.

## Visuals

image

### PR DESCRIPTION
**Pull request by oal55**
# About

* **Related Issues:** #96, #133 

### Changes

- Rewritten the group tests. #96 
- We now have a class based group-operation view. #96
- group-operation route now has a name field. #96
- We can now create a group with empty members & moderators fields. #133 
- Changed moderators and members fields to read only in the group serializer (I saw the other guy said he did this in #137 too, but I couldn't see where he changed it. So I modified the serializer as well :$ ). #133 

Closes #133

### CODE REVIEW

**Comment by karacasoft** [comment]
This has files that are also changed on @Morgazipa 's commits in #137. Prepare for conflicts... 😄

**Review by smddzcy — DISMISSED** [review]
LGTM

**Comment by smddzcy** [comment]
nit: There is a field for related issues in the issue template, please use there for consistency. Issue title is not a good place for this 😄

**Comment by smddzcy** [comment]
Also, again it's not that important but using simple present tense for issue titles is a good practice

**Comment by oal55** [comment]
Thanks for the feedbacks

**Comment by Morgazipa** [comment]
My bad, I should have created an issue for failing tests. Anyway, since our changes are almost identical (except that he introduced different passwords for users) I can revert my changes. But I think that we should change the memberOperation to work with group names ( right now group name is not even an unique field). @triforce930 has raised some other concerns in https://github.com/bounswe/bounswe2017group5/commit/2287f1e4812db4c6fdc31c1bb3f5866bca66938d#diff-a224ae36892ad9b8a65a454b5c43cbf0.

File: api/interestr/api/views.py [hunk 1]
```diff
@@ -157,42 +161,26 @@ class TagDetail(generics.RetrieveUpdateDestroyAPIView):
...
 
-    # get authenticated user.
-    user = request.user
-    if not user.is_authenticated():
-        return HttpResponse(status=403)
-
-    if request.method == 'PUT':
-        if group.members.filter(id=user.id).count() == 1:
-            # handle, user is already a member.
+    def put(self, request, pk):
```
Comment (smddzcy) [inline api/interestr/api/views.py:167]: 👏

**Review by smddzcy — APPROVED** [review]
Looks fine 👌

**Comment by smddzcy** [comment]
I adjusted the description (added a `Closes #??` line) to auto-close the group empty members bug issue, AFAICS this PR fixes that bug. Merge whenever you are ready.

**Comment by oal55** [comment]
Done ~~~

**Pull request merged**